docs/care/CEP/Draft/0010-labs.md

```diff
@@ -0,0 +1,370 @@
+            ("REJECTED", "Sample Rejected"),
+            ("PROCESSING", "Test Ongoing"),
+            ("UNDER_REVIEW", "Result under review"),
+            ("INVALID", "Invalid Result"),
+            ("COMPLETED", "Test Completed"),
+            ("CANCELLED", "Test Cancelled"),
+        ],
+        default="PENDING",
+    )
+    collected_by = models.ForeignKey(
+        User, on_delete=models.PROTECT, null=True, blank=True
+    )
+    collected_date = models.DateTimeField(null=True, blank=True)
+    barcode = models.CharField(max_length=100, unique=True, null=True, blank=True)
+
+    def __str__(self):
+        return f"{self.sample_type} for {self.lab_order}"
+
+
+class LabOrderInvestigation(BaseModel):
+    """
+    Model to store the investigations ordered in the lab order. The investigations can be in different states like Pending, In Progress, Completed, Cancelled.
+    """
+
+    lab_order = models.ForeignKey(
+        LabOrder, on_delete=models.CASCADE, related_name="investigations"
+    )
+    investigation = models.ForeignKey(Investigation, on_delete=models.CASCADE)
+    lab_order_sample = models.ForeignKey(
+        LabOrderSample, on_delete=models.CASCADE, related_name="investigations"
+    )
+    status = models.CharField(
+        max_length=20,
+        choices=[
+            ("PENDING", "Pending"),
+            ("IN_PROGRESS", "In Progress"),
+            ("COMPLETED", "Completed"),
+            ("CANCELLED", "Cancelled"),
+        ],
+        default="PENDING",
+    )
+    notes = models.TextField(blank=True, null=True)
+
+    class Meta:
+        unique_together = ("lab_order", "investigation")
+
+    def __str__(self):
+        return f"{self.investigation.name} for {self.lab_order}"
+
+
+class InvestigationResult(BaseModel):
+    """
+    Model to store the results for the investigations. The results can be in different states like Received, Under Review, Invalid, Accepted.
+    """
+
+    lab_order_investigation = models.ForeignKey(
+        LabOrderInvestigation, on_delete=models.PROTECT, related_name="results"
+    )
+    result_value = models.TextField()
+    is_abnormal = models.BooleanField(default=False)
+    status = models.CharField(
+        max_length=20,
+        choices=[
+            ("RECEIVED", "Result Received"),
+            ("UNDER_REVIEW", "Result under review"),
+            ("INVALID", "Invalid Result"),
+            ("ACCEPTED", "Result Accepted"),
+        ],
+        default="PENDING",
+    )
+    performed_by = models.ForeignKey(
+        User, on_delete=models.PROTECT, related_name="performed_tests"
+    )
+    verified_by = models.ForeignKey(
+        User,
+        on_delete=models.PROTECT,
+        related_name="verified_tests",
+        null=True,
+        blank=True,
+    )
+    notes = models.TextField(blank=True, null=True)
+    custom_min_value = models.FloatField(blank=True, null=True)
+    custom_max_value = models.FloatField(blank=True, null=True)
```

Comment:
Should we have a default min_value and max_value with units? These can override the default ones.

Reply:
Yes, the default min and max values are under investigations, these values can override them when these values are not null